### PR TITLE
Make name, price and what to export configurable

### PR DESCRIPTION
**Pull request by bvogel**

### CODE REVIEW

File: Views/backend/plentymarkets/view/settings.js [hunk 1]
```diff
@@ -252,6 +252,74 @@ Ext.define('Shopware.apps.Plentymarkets.view.Settings', {
...
+                        fieldLabel: '{s name=plentymarkets/view/settings/textfield/ItemFreetextsImportActionID}Freitextfelder übernehmen{/s}',
+                        name: 'ItemFreetextsImportActionID',
+                        xtype: 'checkbox',
+                        inputValue: 1,
+                        uncheckedValue: '0',
+                        supportText: 'Aktivieren, wenn die Freitextfelder von plentymarkets übernommen werden sollen.'
+                    },
+                    {
+                        fieldLabel: '{s name=plentymarkets/view/settings/textfield/ItemPriceImportActionID}Preis übernehmen{/s}',
+                        name: 'ItemNameImportActionID',
```
Comment (ioanamoldovan) [inline Views/backend/plentymarkets/view/settings.js]: name: ItemPriceImportActionID

**Comment by bvogel** [comment]
fixed

**Comment by ioanamoldovan** [comment]
Hi,

at first glance, your implementation seems to work correctly. So far I just found the typing error, which you have already fixed.
I will let you know when the tests are successful and I am ready to merge your pull request.
Thank you for contributing to the source code of our shopware connector.

Best regards,
Ioana

File: Components/Import/Entity/PlentymarketsImportEntityItem.php [hunk 2]
```diff
@@ -220,6 +220,26 @@ protected function setData()
...
+	 *
+	 * @return string
+	 */
+	protected function getItemName($ItemTexts)
+	{
+		$useName = PlentymarketsConfig::getInstance()->getItemNameImportActionID(IMPORT_ITEM_NAME);
+
+		if($useName != 'Name')
+		{
+			if(!empty($ItemTexts->$useName))
```
Comment (ioanamoldovan) [inline Components/Import/Entity/PlentymarketsImportEntityItem.php]: syntax error:  $ItemTexts->$useName

File: Controllers/Backend/Plentymarkets.php [hunk 3]
```diff
@@ -406,6 +406,12 @@ public function saveSettingsAction()
 		$Config->setItemNumberSourceKey($this->Request()->ItemNumberSourceKey);
 		$Config->setItemVariationNumberSourceKey($this->Request()->ItemVariationNumberSourceKey);
 		$Config->setItemBundleHeadActionID($this->Request()->ItemBundleHeadActionID == true ? IMPORT_ITEM_BUNDLE_HEAD : IMPORT_ITEM_BUNDLE_HEAD_NO);
+		$Config->setItemShortDescriptionImportActionID($this->Request()->ItemShortDescriptionImportActionID == true ? IMPORT_ITEM_SHORTDESC : IMPORT_ITEM_SHORTDESC_NO);
+		$Config->setItemLongDescriptionImportActionID($this->Request()->ItemShortDescriptionImportActionID == true ? IMPORT_ITEM_LONGDESC : IMPORT_ITEM_LONGDESC_NO);
+		$Config->setItemKeywordsImportActionID($this->Request()->ItemKeywordsImportActionID == true ? IMPORT_ITEM_KEYWORDS : IMPORT_ITEM_KEYWORDS_NO);
+		$Config->setItemNameImportActionID($this->Request()->ItemNameImportActionID);
+		$Config->setItemPriceImportActionID($this->Request()->ItemPriceImportActionID);
+		$Config->setItemFreetextsImportActionID($this->Request()->ItemFreetextsImportActionID == true ? IMPORT_ITEM_FREETEXTS : IMPORT_ITEM_FREETEXTS_NO);
```
Comment (ioanamoldovan) [inline Controllers/Backend/Plentymarkets.php]: ItemFreetextsImportActionID ist beim Import von Freitextfelder in setDetails() Methode nicht berücksitigt. Es müsste noch eingebaut werden , wenn der Import von Freitextfelder auch konfigurierbar sein sollte.

**Comment by ioanamoldovan** [comment]
Hi,
Wenn der Import von Name und  Beschreibungen des Artikels konfigurierbar sein müsste, dann müsste die Übersetzungen auch dementsprechend angepasst werden.

Schöne Grüße
Ioana

File: Components/Import/Entity/PlentymarketsImportEntityItemPrice.php [hunk 4]
```diff
@@ -179,6 +179,26 @@ public function getPrices()
...
+	 */
+	protected function getItemPrice($ItemPrices)
+	{
+		$usePrice = PlentymarketsConfig::getInstance()->getItemPriceImportActionID(IMPORT_ITEM_PRICE);
+
+		if($usePrice != 'Price')
+		{
+			if(!empty($ItemPrices->$usePrice))
+			{
+				return $ItemPrices->$usePrice;
```
Comment (ioanamoldovan) [inline Components/Import/Entity/PlentymarketsImportEntityItemPrice.php]: syntax error : $ItemPrices->$usePrice

**Comment by bvogel** [comment]
@ioanamoldovan wir setzen diesen patch jetzt erfolgreich in unserem Produktivsystem ein. Was müssten wir noch tun damit er gemergt wird?

**Comment by jochenmanz** [comment]
Hi,

i really would like to merge your changes for the next major release. That release will include Shopware 5.2.x compatibility. Could you check your changes against the development branch and a Shopware 5.2 installation if they still work as expected?

Best regards,
Jochen

**Comment by bvogel** [comment]
@jochenmanz unfortunately we are not in the position to test this change against a 5.2 system as we would need a 5.2 test system. We currently do not dispose neither of time nor resources to setup a 5.2 system. This may be possible in Q1 2017, but as all the functionality is only dependent from the plugin in itself and not the shopware version it should be working as expected at least against the API 115.

**Pull request merged**